### DISCUSSION TITLE
Control position independent of object angle?

### DISCUSSION THREAD

**Discussion by DatCaptainHorse** [fabricjs/fabric.js#7649] [General]
Just wondering if this is currently possible, if you want controls to follow object's position but not rotation.
I suppose it's possible with custom render functions?

**Comment by asturur** [comment]
I think is possible, but not for the control box.
So you should disable the drawing of the  border with `hasBorders: false` at least.
Then you would need a good wrapper for the render functions to counter rotate the object.

**Reply by DatCaptainHorse** [reply]
Custom render functions work for the visual side, but the clickable position of control still follows rotation. Is there a way to override that as well in Fabric?

**Reply by asturur** [reply]
sorry i did not think of this issue.

**Comment by DatCaptainHorse** [comment]
I've found a solution, noticed there is a `positionHandler` function for Control, I overrode that with modified function and forced the matrices' rotation-affected elements to identity, see below:

```
function(dim, final) {
    // Get point
    const p = new fabric.Point(this.x * dim.x + this.offsetX, this.y * dim.y + this.offsetY);
    // Modify final matrix by setting rotation-affected elements to identity
    final[0] = 1;
    final[1] = 0;
    final[2] = 0;
    final[3] = 1;
    // Transform point by matrix
    return fabric.util.transformPoint(p, final);
}
```

This makes controls be unaffected by the rotation of the object. Note that I'm still using Fabric.js 4.6.0, this may not work with the new 5.0 version.

**Reply by asturur** [reply]
5.0 did not. change anything on that side

**Reply by asturur** [reply]
Yes position handler is definitely the api part to do your own logic, the one provided just mirror the basic functionality of fabricjs
